### DISCUSSION TITLE
iperf3 UDP server

### DISCUSSION THREAD

**Discussion by piddalagi** [esnet/iperf#1319] [Q&A]
Dear Team,

from iperf3 onwards, we should just mentioned "iperf3 -s" to listen on default port of 5201.
netstat shows that server is listening on tcp6.

Please help me understand how are we handling UDP client packets, when server socket is of type stream.
Any basic pointers from code perspective is appreciated.

_Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp6       0      0 :::5201                 :::*                    LISTEN      12743/iperf3          
root@pidda:/home/pidda/Documents/MyPHD/containernet/examples#_

**Comment by bmah888** [comment]
iperf3 (unlike a number of other utilities) uses a control connection, which runs over TCP, to exchange data between the client and server before the start of the actual data transfer. The socket you are seeing on the server is listening for the client to make the initial TCP control connection.

**Reply by piddalagi** [reply]
thanks Bruce. Got it.
